### PR TITLE
feat: use sdkv3

### PR DESCRIPTION
**Pull request by TheRealAmazonKendra**
This updates usage of the sdk to sdkv3 and begins publishing it on a new major version. For the moment we will publish these as release candidates until we are sure of stability.

### CODE REVIEW

File: lib/aws.ts [hunk 1]
```diff
@@ -43,65 +61,75 @@ export interface Account {
...
+  public async upload(params: PutObjectCommandInput, options: ClientOptions): Promise<CompleteMultipartUploadCommandOutput> {
+    try {
+      const upload = new Upload({
+        client: await this.s3Client(options),
+        params,
+      });
+
+      return upload.done();
+    } catch (e) {
+      // TODO: add something more useful here
```
Comment (comcalvi) [inline lib/aws.ts:94]: todo?

File: lib/aws.ts [hunk 2]
```diff
@@ -111,48 +139,24 @@ export class DefaultAwsClient implements IAws {
...
+        config.credentials = fromTemporaryCredentials({
+          params: {
+            RoleArn: options.assumeRoleArn,
+            ExternalId: options.assumeRoleExternalId,
+            RoleSessionName: `${USER_AGENT}-${safeUsername()}`,
+          },
+          clientConfig: this.config.clientConfig,
+        });
+      }
+      return config;
```
Comment (comcalvi) [inline lib/aws.ts:157]:
```suggestion
```

File: lib/aws.ts [hunk 3]
```diff
@@ -111,48 +139,24 @@ export class DefaultAwsClient implements IAws {
...
-        RoleArn: roleArn,
-        ExternalId: externalId,
-        RoleSessionName: `cdk-assets-${safeUsername()}`,
-      },
-      stsConfig: {
-        region,
-        customUserAgent: 'cdk-assets',
-      },
-    });
+    return this.config;
```
Comment (comcalvi) [inline lib/aws.ts:159]:
```suggestion
    return config;
```

File: lib/private/handlers/container-images.ts [hunk 4]
```diff
@@ -249,12 +249,17 @@ class ContainerImageBuilder {
...
+    const command = new DescribeImagesCommand({
+      repositoryName,
+      imageIds: [{ imageTag }],
+    });
+
+    await ecr.send(command);
     return true;
   } catch (e: any) {
-    if (e.code !== 'ImageNotFoundException') {
+    if (e.name !== 'ImageNotFoundException') {
```
Comment (comcalvi) [inline lib/private/handlers/container-images.ts:262]: why the change from `code` to `name`?

File: lib/private/handlers/files.ts [hunk 5]
```diff
@@ -130,7 +131,7 @@ export class FileAssetHandler implements IAssetHandler {
       paramsEncryption
     );
 
-    await s3.upload(params).promise();
+    await this.host.aws.upload(params, {});
```
Comment (comcalvi) [inline lib/private/handlers/files.ts:134]: can we make the second parameter optional?

File: lib/aws.ts (hunk 1 above)
Comment (TheRealAmazonKendra) [reply lib/aws.ts:94]: I'll add something here before I remove the release candidate tag. I need to actually test the failure modes in the CLI to figure out what will be useful information here.

File: lib/private/handlers/container-images.ts (hunk 4 above)
Comment (TheRealAmazonKendra) [reply lib/private/handlers/container-images.ts:262]: SDK v3 updated the error class as such that code is no longer a field and name is the replacement for it. No idea why they did that ¯\_(ツ)_/¯

**Review by comcalvi — APPROVED** [review]
